de-lib/src/main/java/org/iplantc/de/apps/client/views/details/AppDetailsViewImpl.java

```diff
@@ -230,24 +235,63 @@ public void setBaseDebugId(final String baseId) {
             url.addClickHandler(new ClickHandler() {
                 @Override
                 public void onClick(ClickEvent event) {
-                    IPlantPromptDialog ipd = new IPlantPromptDialog(appearance.appUrl(),
-                                                                    1024,
-                                                                    GWT.getHostPageBaseURL()
-                                                                            + "?type="
-                                                                            + DesktopPresenterImpl.TypeQueryValues.APPS
-                                                                            + "&app-id=" + app.getId(),
-                                                                    null);
+                    Dialog ipd = new Dialog();
+                    ipd.setModal(true);
+                    ipd.setHideOnButtonClick(true);
                     ipd.setHeadingHtml(appearance.copyAppUrl());
                     ipd.setWidth("500px");
                     ipd.setPredefinedButtons(PredefinedButton.OK);
                     ipd.show();
+                    renderCopyTextArea(ipd.getBody().getId(true),
+                                       appearance.copyAppUrl(),
+                                       GWT.getHostPageBaseURL()
+                                                    + "?type="
+                                                    + DesktopPresenterImpl.TypeQueryValues.APPS
+                                                    + "&app-id=" + app.getId());
                 }
             });
         } else {
             helpLink.setVisible(false);
         }
+
+        Scheduler.get().scheduleFinally(new Scheduler.ScheduledCommand() {
+
+            @Override
+            public void execute() {
+                renderToolDetails(toolsContainer.getId(),
+                                  getAppDetailsAppearanceShim(appearance),
+                                  AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(app)));
+            }
+        });
     }
 
+    public static native void renderCopyTextArea(String elementID, String btnText, String textToCopy) /*-{
+        $wnd.CyVerseReactComponents.renderCopyTextArea(elementID, btnText, textToCopy);
+    }-*/;
+
+    public static native void renderToolDetails(String elementID, JavaScriptObject appearance, Splittable app) /*-{
+        $wnd.CyVerseReactComponents.renderToolDetails(elementID, appearance, app);
+    }-*/;
+
+    public static native JavaScriptObject getAppDetailsAppearanceShim(AppDetailsView.AppDetailsAppearance appearance) /*-{
```

Comment:
This `shim` method was required to box the appearance method calls into a `JavaScriptObject` that could be passed to the React component.

If we upgrade to GWT 2.8, then I think the appearance interface could be annotated as a `@JsType` and the appearance could be passed directly to the React components without this `JavaScriptObject` shim.